### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by cosmas-mnt**
* Migrate Lightning dependency to 2.x
* Define project dependencies in requirements.txt
* Remove unused imports
* Minor bug fixes

### CODE REVIEW

**Comment by typiconman** [comment]
@pgmmpk please review and approve.

**Comment by cosmas-mnt** [comment]
I am planning another contribution about refactoring data preparation into a pipeline based on Apache Beam.

What do you think?

**Comment by pgmmpk** [comment]
@cosmas-mnt 

what were the bugs, and how can I see the fix? Could you maybe split bugfixes out of this big cleanup PR?

File: accentru/accentru_dataset.py [hunk 1]
```diff
@@ -168,7 +166,7 @@ def __getitem__(self, i):
 
     args = parser.parse_args()
 
-    data = AccentDataset(max_len=args.max_len, batch_size=args.batch_size)
+    data = AccentruDataset(max_len=args.max_len, batch_size=args.batch_size)
```
Comment (cosmas-mnt) [inline accentru/accentru_dataset.py:169]: BUG: Reference to a dataset not defined in current file.

File: translator/onnx_export.py [hunk 2]
```diff
@@ -60,14 +59,8 @@ def to_onnx(model_checkpoint='model.ckpt', onnx_filename='model.onnx'):
     print(torch_logits)
 
 def sample(onnx_model, words=['лепота', 'несть']):
-
     session = ort.InferenceSession('model.onnx')
 
-    logits = session.run(None, {
-        'inputs': inputs.numpy(),
-        'accents': accents.numpy(),
-    })
```
Comment (cosmas-mnt) [inline translator/onnx_export.py:69]: BUG: Reference to `inputs` and `accents` before they were created.

Exactly the same code snippet exists after variables' initialization.

File: translator/translator_dataset.py [hunk 3]
```diff
@@ -196,7 +194,7 @@ def __getitem__(self, i):
 
     args = parser.parse_args()
 
-    data = TranslatorDataset6(max_len=args.max_len, batch_size=args.batch_size)
+    data = TranslatorDataset(max_len=args.max_len, batch_size=args.batch_size)
```
Comment (cosmas-mnt) [inline translator/translator_dataset.py:197]: BUG: Reference to a class that doesn't exist.

**Comment by cosmas-mnt** [comment]
@pgmmpk I put comments on the 3 bugs I found. I think they are trivial.

Would you still recommend split commit?

**Comment by pgmmpk** [comment]
All looks good. Thanks for the cleanup effort. Merged!

**Pull request merged**